hazelcast/src/main/java/com/hazelcast/cache/impl/HazelcastServerCachingProvider.java

```diff
@@ -133,8 +134,8 @@ private Config getConfigFromLocation(URI location, ClassLoader classLoader, Stri
     }
 
     private Config getConfig(URL configURL, ClassLoader theClassLoader, String instanceName) throws IOException {
-        Config config = new XmlConfigBuilder(configURL).build()
-                .setClassLoader(theClassLoader);
+        Config config = Config.loadFromFile(new File(configURL.getPath()));
```

Comment:
this works fine as long as the URL represents a local `file:` or `classpath:` resource. However it breaks the use case for `http:` / `https:` (and probably `jar:` schemes... so after all it looks like we shouldn't use the `Config.loadFromFile` facility and make a guess based on `.xml`/`.yaml`/`.yml` extension of the URL path.